### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by stevebaxter**
This fixes all the tests and guard warnings. I'm sure that Hound will have something to bark about.

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "overrides": [
```
Comment (stevebaxter) [inline .prettierrc]: This sets Prettier (https://github.com/prettier/prettier-vscode) up to match the rules in .scss-lint.yml. Nice of you use VS Code like me.

File: app/assets/stylesheets/home.scss [hunk 2]
```diff
@@ -6,35 +6,35 @@
 @import "normalize-rails/normalize";
 
 body {
-  font-family: "Open Sans", sans-serif;
-  margin: 0;
-  font-size: $base-font-size;
-  line-height: $base-line-height;
-  background-color: $bg-color; // For "infinite" footer
+    font-family: "Open Sans", sans-serif;
```
Comment (stevebaxter) [inline app/assets/stylesheets/home.scss]: No changes here, just updating the spacing.

File: spec/controllers/user/swaps_controller_spec.rb [hunk 3]
```diff
@@ -21,9 +23,10 @@
     let(:an_email) { double(:an_email) }
 
     before do
-      session[:user_id] = new_user.id
-      allow(User).to receive(:find_by_id).with(new_user.id)
-                       .and_return(new_user)
+      # Stub out authentication
+      allow(request.env["warden"]).to receive(:authenticate!).and_return(new_user)
```
Comment (stevebaxter) [inline spec/controllers/user/swaps_controller_spec.rb]: Cunning, see https://github.com/plataformatec/devise/wiki/How-To:-Stub-authentication-in-controller-specs

File: spec/factories/user.rb [hunk 4]
```diff
@@ -1,6 +1,18 @@
 FactoryBot.define do
   factory :user do
-    name { "John Doe" }
-    email { "john@example.com" }
+    name { "John" }
+    email { "#{name.gsub(/\s/, ".").downcase}@example.com" }
```
Comment (stevebaxter) [inline spec/factories/user.rb]: Made this a bit more sophisticated so it's easier to create tests

File: spec/helpers/user_helper_spec.rb [hunk 5]
```diff
@@ -6,9 +6,9 @@
     let(:name) { "Alice Bloggs" }
     let(:user) { create(:user, name: name) }
 
-    it "returns just the name when there is no profile URL" do
+    it "returns email link when there is no profile URL" do
       user.create_identity(provider: :facebook)
-      expect(helper.user_profile_link(user)).to eq name
+      expect(helper.user_profile_link(user)).to eq "<a href=\"mailto://alice.bloggs@example.com\">Alice Bloggs</a>"
```
Comment (stevebaxter) [inline spec/helpers/user_helper_spec.rb]: This is a change in behaviour that I think we want @jpallen ?

File: spec/models/user_spec.rb [hunk 6]
```diff
@@ -14,7 +16,7 @@
 
     context "with user with constituency id" do
       let(:constituency) { OnsConstituency.create!(name: "test con 1", ons_id: "another-fake-ons-id") }
-      let(:user) { User.new(email: "test@example.com", name: "test user", constituency_ons_id: constituency.ons_id) }
+      let(:user) { create(:user, name: "Test", constituency_ons_id: constituency.ons_id) }
```
Comment (stevebaxter) [inline spec/models/user_spec.rb]: Use our factory so things are managed from one place

File: spec/models/user_spec.rb [hunk 7]
```diff
@@ -143,32 +145,6 @@
   end
 
   describe "#send_welcome_email (after_save callback)" do
-    context "when the user does NOT have an email address" do
```
Comment (stevebaxter) [inline spec/models/user_spec.rb]: I don't think this applies anymore as users will always have to have an email address @jpallen ?

File: spec/controllers/users_controller_spec.rb [hunk 8]
```diff
@@ -5,22 +5,25 @@
...
+      # Stub out authentication
+      allow(request.env["warden"]).to receive(:authenticate!).and_return(logged_in_user)
+      allow(controller).to receive(:current_user).and_return(logged_in_user)
     end
 
     describe "GET #show" do
       it "returns http success" do
-        swap_with_user = create(:user)
-        logged_in_user.swap_with_user_id(swap_with_user.id)
+        ClimateControl.modify SERVER_HOST_PORT: "localhost:3000" do
```
Comment (stevebaxter) [inline spec/controllers/users_controller_spec.rb]: This looks like a nicer way of controlling the test environment, see https://thoughtbot.com/blog/testing-and-environment-variables

**Comment by stevebaxter** [comment]
OK, modifying the test environment isn't quite working, closing while I fix this.

**Comment by stevebaxter** [comment]
OK, this all passes. @jpallen - please could you merge this into your branch when you're ready?

File: spec/helpers/user_helper_spec.rb (hunk 5 above)
Comment (jpallen) [reply spec/helpers/user_helper_spec.rb]: Hmm, yes, partly. I hadn't seen the `user_profile_link` helper. I think we can be a little cleverer there. This is good for now though, thanks!

File: spec/models/user_spec.rb (hunk 7 above)
Comment (jpallen) [reply spec/models/user_spec.rb]: 👍

**Comment by jpallen** [comment]
🎉 You are my current personal Person Of The Month @stevebaxter, this is amazing thank you.

**Pull request merged**